Dockerfile

```diff
@@ -1,81 +1,97 @@
+# Dockerfile containerizes the Next.js apps.
+# https://pnpm.io/docker#example-2-build-multiple-docker-images-in-a-monorepo
 
+# alpine is a minimal build of Node. Here it's used to build the FEM libraries and then the Next.js apps.
+FROM node:22-alpine AS base
+
+# Setup pnpm
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+
+FROM base AS builder
+
+# Get access to the commit id.
 ARG COMMIT_ID
 ENV COMMIT_ID=$COMMIT_ID
 ENV SENTRY_RELEASE=$COMMIT_ID
 
+# Get access to the branch name.
 ARG GITHUB_REF_NAME
 ENV GITHUB_REF_NAME=$GITHUB_REF_NAME
 
-ARG PANOPTES_ENV=production
-ENV PANOPTES_ENV=$PANOPTES_ENV
-
-ARG NODE_ENV=production
-ENV NODE_ENV=$NODE_ENV
-
-ARG APP_ENV=production
-ENV APP_ENV=$APP_ENV
-
+# Opt out of Next.js telemetry.
 ENV NEXT_TELEMETRY_DISABLED=1
 
+# Add tokens for Contentful and Sentry.
 ARG CONTENTFUL_ACCESS_TOKEN
-
 ARG CONTENTFUL_SPACE_ID
-
 ARG SENTRY_AUTH_TOKEN
 
+# Set the working directory. -p will create all the necessary parent directories along the way.
 RUN mkdir -p /usr/src
-
 WORKDIR /usr/src/
 
-ADD package.json /usr/src/
-
-COPY .yarn /usr/src/.yarn
+# Add the root package.json.
+COPY package.json /usr/src/
 
-ADD .yarnrc /usr/src/
+# Add the pnpm config.
+COPY .npmrc /usr/src/
+COPY pnpm-workspace.yaml /usr/src/
 
-ADD lerna.json /usr/src/
+# Add the lock file.
+COPY pnpm-lock.yaml /usr/src/
 
+# Copy over the code in /packages.
 COPY ./packages /usr/src/packages
 
-ADD yarn.lock /usr/src/
-
+# Build as `USER: node` instead of the default `root` user https://github.com/nodejs/docker-node/blob/main/docs/BestPractices.md#non-root-user.
 RUN chown -R node:node .
-
 USER node
 
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn install --production=false --frozen-lockfile --ignore-scripts
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn workspace @zooniverse/react-components build:es6
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn workspace @zooniverse/subject-viewers build:es6
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn workspace @zooniverse/content build:es6
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn workspace @zooniverse/classifier build:es6
-RUN --mount=type=cache,id=fem-builder-yarn,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn workspace @zooniverse/user build:es6
+# Use BuildKit cache mounts to share the pnpm store between builds.
+# Include devDep in this step because Babel is needed to build the Zooniverse libraries.
+# The prepare scripts in each package run now too.
+RUN --mount=type=cache,target=/pnpm/store pnpm install --frozen-lockfile
```

Comment:
`pnpm install` without including `--ignore-scripts` is a more concise way to install dependencies and then immediately build the Zooniverse libraries in FEM. In some cases the `prepare` scripts still include `build cjs`, but I'm going to remove the CJS builds from all Zooniverse libraries, so this Dockerfile change is planning ahead.